### PR TITLE
DIV-5049 - Add address input for respondent solicitor on RFE

### PR DESCRIPTION
**Pull request by dtaylor7**
# Description

[DIV-5049 - Add address input for respondent solicitor on RFE](https://tools.hmcts.net/jira/browse/DIV-5049)
 - Added config for postcode url and token for each environments
 - Added postcode lookup service and middleware
 - Added postcode lookup pages
 - Added manual address input page
 - Added confirm address page
 - Check you answers values
 - Unit and E2E tests
 - Mocks for postcode lookup

### CODE REVIEW

File: steps/respondent/solicitor-address/SolicitorAddress.step.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    return { selectedAddress };
+  }
+
+  get editAddressForm() {
+    const address = list(text);
+    return { address };
+  }
+
+  get confirmAddressForm() {
+    const confirmAddressAnswers = ['yes', 'no'];
```
Comment (alexwakeman) [inline steps/respondent/solicitor-address/SolicitorAddress.step.js]: We really should extract these 'yes' 'no' values into a constants file for the project - but for now we could extract into a variable maybe..

File: steps/respondent/solicitor-address/templates/address.html [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+{% from "look-and-feel/components/header.njk" import header %}
+{% from "look-and-feel/components/phase_banner.njk" import phaseBanner %}
+{% from "look-and-feel/components/errors.njk" import errorSummary %}
+{% from "look-and-feel/components/i18n.njk" import languageSwitch %}
+
+{% from "look-and-feel/components/fields.njk" import selectionButtons, textbox, errorClass, errorsFor, select, textarea %}
+{% from "select/macro.njk" import govukSelect %}
+
+{% set defaultAddressContent = {
+    pageTitle: pageContent.pageTitle if pageContent.pageTitle else 'What is the address?',
```
Comment (alexwakeman) [inline steps/respondent/solicitor-address/templates/address.html]: `pageTitle` is static right? Why do we need a conditional on it?

Comment (dtaylor7) [reply steps/respondent/solicitor-address/templates/address.html]: its just setting a default encase no title configured in steps/respondent/solicitor-address/SolicitorAddress.html

File: steps/respondent/solicitor-address/SolicitorAddress.step.js (hunk 1 above)
Comment (dtaylor7) [reply steps/respondent/solicitor-address/SolicitorAddress.step.js]: 👍

File: yarn.lock [hunk 3]
```diff
@@ -5602,7 +5602,7 @@ media-typer@0.3.0:
   resolved "https://registry.yarnpkg.com/media-typer/-/media-typer-0.3.0.tgz#8710d7af0aa626f8fffa1ce00168545263255748"
   integrity sha1-hxDXrwqmJvj/+hzgAWhUUmMlV0g=
 
-mem@^1.1.0, mem@^4.0.0, "mem@^4.0.0 ":
+mem@^1.1.0, mem@^4.0.0:
```
Comment (yarpo) [inline yarn.lock]: is there any chance we can easily get rid of mem 1.1.0?

Comment (dtaylor7) [reply yarn.lock]: 👍

File: infrastructure/variables.tf [hunk 4]
```diff
@@ -206,3 +206,7 @@ variable "evidence_management_download_endpoint" {
 variable "feature_resp_solicitor_details" {
   default = false
 }
+
+variable "post_code_url" {
+  default = "https://api.ordnancesurvey.co.uk/places/v1"
```
Comment (hosainnet) [inline infrastructure/variables.tf]: If we're defaulting in here not sure we need to put it in every tfvars file above

Comment (dtaylor7) [reply infrastructure/variables.tf]: 👍

File: yarn.lock (hunk 3 above)
Comment (dtaylor7) [reply yarn.lock]: apparently not https://app.snyk.io/org/hmcts/test/github/db8c3906-75ab-45fc-8f08-d542391d30e5/13a875395f25e5c3079e21519fda1208ee8639fb..c954edc0221aedcb461408698ff9df45c8c79226?fromStatus=true

**Comment by dtaylor7** [comment]
closing until ready to be tested

**Comment by dtaylor7** [comment]
closing until we decide this is to be merged

**Comment by hosainnet** [comment]
@dtaylor7 this has been signed-off no? It's required for the release in a couple of weeks and it's hidden behind a feature flag so I think we should merge it to be code complete - @gavinclark can confirm

**Comment by dtaylor7** [comment]
> @dtaylor7 this has been signed-off no? It's required for the release in a couple of weeks and it's hidden behind a feature flag so I think we should merge it to be code complete - @gavinclark can confirm

it has many conflicts and believe some issues, been sat here for a while so will need to be retested.

**Comment by hosainnet** [comment]
> > @dtaylor7 this has been signed-off no? It's required for the release in a couple of weeks and it's hidden behind a feature flag so I think we should merge it to be code complete - @gavinclark can confirm
> 
> it has many conflicts and believe some issues, been sat here for a while so will need to be retested.

@dtaylor7 but why close it? it is to be merged..

**Pull request merged**